python/ray/util/data/parquet.py

```diff
@@ -0,0 +1,92 @@
+import random
+from typing import Iterable
+from typing import List, Optional, Union
+
+import pyarrow.parquet as pq
+from pandas import DataFrame
+
+import ray.util.iter as para_iter
+from .dataset import MLDataset
+from .source import SourceShard
+
+
+class ParquetSourceShard(SourceShard):
+    def __init__(self,
+                 data_pieces: List[pq.ParquetDatasetPiece],
+                 columns: Optional[List[str]],
+                 partitions: Optional[pq.ParquetPartitions],
+                 shard_id: int):
+        self._data_pieces = data_pieces
+        self._columns = columns
+        self._partitions = partitions
+        self._shard_id = shard_id
+
+    @property
+    def shard_id(self) -> int:
+        return self._shard_id
+
+    def __iter__(self) -> Iterable[DataFrame]:
+        for piece in self._data_pieces:
+            yield piece.read(columns=self._columns, use_threads=False,
+                             partitions=self._partitions).to_pandas()
+
+
+def read_parquet(paths: Union[str, List[str]],
```

Comment:
This is pretty cool. How about we add a small example on how to use this, e.g.

```
# create dummy data
spark.range(...).write.parquet(...)
# create mldataset
data = ray.util.data.read_parquet(...)
# feed into training actors
[actors[n].set_input(data.get_shard(i)) for i in range(n)]
```

Maybe @richardliaw has more ideas for how to integrate with existing training examples?

Reply:
Yeah, you could try making this work with TorchTrainer for Pytorch. https://docs.ray.io/en/master/raysgd/raysgd_dataset.html#complete-dataset-example

Alternatively, we'll have xgboost-ray distributed file reading ready soon, so that'll be also a good integration point.

Reply:
Thanks for the advice. I will add it and cover the UT.